examples/plot_barber2020_simulations.py

```diff
@@ -103,61 +123,75 @@ def plot_simulation_results(results: Dict, methods: List[str], title: str) -> No
     """
     fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 5))
-    plt.rcParams.update({'font.size': 14})
+    plt.rcParams.update({"font.size": 14})
     if title is not None:
         plt.suptitle(title)
     for method in methods:
-        coverage_mean = []
-        coverage_SE = []
-        for dim_val in dim_vals:
+        coverage_mean, coverage_SE = [], []
+        width_mean, width_SE = [], []
+        for dimension in dimensions:
             coverage_mean.append(np.array([
-                results[key][3] for key in results
-                if (results[key][2] == method) & (results[key][1] == dim_val)
+                results[key]["coverage"] for key in results
+                if (
+                    (results[key]["method"] == method) &
+                    (results[key]["dimension"] == dimension)
+                )
             ]).mean())
             coverage_SE.append(np.array([
-                results[key][3] for key in results
-                if (results[key][2] == method) & (results[key][1] == dim_val)
-            ]).std()/np.sqrt(ntrial))
-        coverage_mean = np.array(coverage_mean)
-        coverage_SE = np.array(coverage_SE)
-        ax1.plot(dim_vals, coverage_mean, label=method)
-        ax1.fill_between(
-            dim_vals, coverage_mean-coverage_SE, coverage_mean+coverage_SE, alpha=0.25
-        )
-    ax1.axhline(1-alpha, linestyle='dashed', c='k')
-    ax1.set_ylim(0.0, 1.0)
-    ax1.set_xlabel('Dimension d')
-    ax1.set_ylabel('Coverage')
-    ax1.legend()
-    for method in methods:
-        width_mean = []
-        width_SE = []
-        for dim_val in dim_vals:
+                results[key]["coverage"] for key in results
+                if (
+                    (results[key]["method"] == method) &
+                    (results[key]["dimension"] == dimension)
+                )
+            ]).std(ddof=np.sqrt(ntrial)))
             width_mean.append(np.array([
-                results[key][-1] for key in results
-                if (results[key][2] == method) & (results[key][1] == dim_val)
+                results[key]["width_mean"] for key in results
+                if (
+                    (results[key]["method"] == method) &
+                    (results[key]["dimension"] == dimension)
+                )
             ]).mean())
             width_SE.append(np.array([
-                results[key][-1] for key in results
-                if (results[key][2] == method) & (results[key][1] == dim_val)
-            ]).std()/np.sqrt(ntrial))
-        width_mean = np.array(width_mean)
-        width_SE = np.array(width_SE)
-        ax2.plot(dim_vals, width_mean, label=method)
+                results[key]["width_mean"] for key in results
+                if (
+                    (results[key]["method"] == method) &
+                    (results[key]["dimension"] == dimension)
+                )
+            ]).std(ddof=np.sqrt(ntrial)))
+        coverage_mean_np = np.stack(coverage_mean)
```

Comment:
Si tu n'utilises plus la version liste ensuite, tu peux substituer in-place 
```
coverage_mean = np.stack(coverage_mean)

Reply:
A la base c'est ce que j'avais fait mais mypy me donne ce message d'erreur:
`error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[number[Any]]")`

Reply:
Ah oui je vois, c'est parce que tu switches de list à array. Pas moyen d'être en array dès le départ ?

Reply:
J'ai ajouté un `np.zeros(shape)`.